pkg/controller/intrusiondetection/intrusiondetection_controller.go

```diff
@@ -310,6 +334,40 @@ func (r *ReconcileIntrusionDetection) Reconcile(ctx context.Context, request rec
 		return reconcile.Result{}, err
 	}
 
+	// Query for StorageClass for AD API if provided
+	shouldRenderADPVC := false
+	if !isManagedCluster && instance.Spec.AnomalyDetection.StorageType == operatorv1.PersistentStorageType {
+		// validate to degrade early if the storage class name is not valid
+		if err = utils.ValidateResourceNameIsQualified(instance.Spec.AnomalyDetection.StorageClassName); err != nil {
+			errMessage := "invalid Anomaly Detection Storage Class name provided"
+			r.SetDegraded(operatorv1.InvalidConfigurationError, errMessage, err, reqLogger)
+			return reconcile.Result{}, err
+		}
+
+		if err = r.client.Get(ctx, client.ObjectKey{Name: instance.Spec.AnomalyDetection.StorageClassName}, &storagev1.StorageClass{}); err != nil {
+			if errors.IsNotFound(err) {
+				errMessage := "failed to get storage class for anomaly detection"
+				r.SetDegraded(operatorv1.ResourceNotFound, errMessage, err, reqLogger)
+				return reconcile.Result{}, err
+			} else {
+				errMessage := "failed to query for Anomaly Detection Storage Class"
+				r.SetDegraded(operatorv1.InternalServerError, errMessage, err, reqLogger)
+				return reconcile.Result{}, err
+			}
+		}
+
+		err = r.client.Get(ctx, client.ObjectKey{Name: render.ADPersistentVolumeClaimName, Namespace: render.IntrusionDetectionNamespace}, &corev1.PersistentVolumeClaim{})
```

Comment:
Your comment in the render says that PVCs are immutable. So if we find one I think we should be checking that the storageClassName matches what is in the PVC and if not the PVC should be removed so a new one can be created. I'm not sure if that also means the deployment (or whatever resource) that uses the PVC might need to be stopped first too.

Reply:
I still have a concern that there may be some setups that will block us from being able to clean up the PVC with a deployment still referencing it, but that is probably an edge case that we might not need to worry about.